File: LATIHAN_ARRAY/js/c3.js
function manageStack(arr, operations) {
    const stack = [...arr]; // salin array awal agar tidak berubah
    const log = [];

    for (let i = 0; i < operations.length; i++) {
        const operasi = operations[i];

        if (operasi.startsWith("push:")) {
            let nilai = operasi.split(":")[1];

            // ubah string angka jadi number jika memungkinkan
            if (!isNaN(Number(nilai))) {
                nilai = Number(nilai);
            }

            stack.push(nilai);
            log.push(`Added ${nilai}`);
        } else if (operasi === "pop") {
            const nilaiDihapus = stack.pop();
            log.push(`Removed ${nilaiDihapus}`);
        }
    }

    return {
        result: stack,
        log: log,
    };
}

// driver code
console.log(manageStack([1, 2], ["push:3", "push:4", "pop"]));
console.log(manageStack([], ["push:a", "push:b", "pop", "push:c"]));

File: LATIHAN_ARRAY/rs/src/c_array/c3.rs
#[derive(Debug)]
pub struct StackResult {
    pub result: Vec<String>,
    pub log: Vec<String>,
}

pub fn manage_stack(
    arr: Vec<String>,
    operations: Vec<&str>,
) -> StackResult {
    let mut stack = arr.clone();
    let mut log: Vec<String> = Vec::new();

    for operasi in operations {
        if operasi.starts_with("push:") {
            let value = operasi[5..].to_string();
            stack.push(value.clone());
            log.push(format!("Added {}", value));
        } else if operasi == "pop" {
            if let Some(removed) = stack.pop() {
                log.push(format!("Removed {}", removed));
            }
        }
    }

    StackResult { result: stack, log }
}
